radixdlt/src/integration/java/com/radixdlt/consensus/CrashFaultNetworkTest.java

```diff
@@ -17,52 +17,203 @@
+			.map(EventCoordinatorNetworkRx::proposalMessages)
+			.collect(Collectors.toList());
+		Observable<Object> directProposalsCheck = Observable.merge(correctProposals)
+			.filter(v -> correctNodesPubs.contains(bftNetwork.getProposerElection().getProposer(v.getView().previous())))
+			.doOnNext(v -> assertThat(v)
+				.satisfies(new Condition<>(vtx -> vtx.getView().equals(vtx.getParentView().next()),
+					"Vertex after correct %s at %s has direct parent",
+					bftNetwork.getProposerElection().getProposer(v.getParentView()).euid(), v.getParentView())))
+			.map(o -> o);
+		Observable<Object> gapProposalsCheck = Observable.merge(correctProposals)
+			.filter(v -> !correctNodesPubs.contains(bftNetwork.getProposerElection().getProposer(v.getView().previous())))
+			.doOnNext(v -> assertThat(v)
+				.satisfies(new Condition<>(vtx -> !vtx.getView().equals(vtx.getParentView().next()),
+					"Vertex after faulty %s at %s has gap",
+					bftNetwork.getProposerElection().getProposer(v.getParentView()).euid(), v.getParentView())))
+			.map(o -> o);
+
+		Observable.mergeArray(bftNetwork.processBFT(), correctCommitCheck, correctTimeoutCheck, directProposalsCheck, gapProposalsCheck)
+			.take(time, timeUnit)
+			.blockingSubscribe();
+	}
+
+	/**
+	 * Tests a dynamic configuration of 7 nodes (all correct at start),
+	 * where correct nodes are crash-stopped at random over time.
+	 * The intended behaviour is that the correct nodes keep making progress until QCs can no longer be formed.
+	 */
+	@Test
+	public void given_7_correct_bfts_that_randomly_crash__then_correct_instances_should_make_progress_as_possible_over_1_minute() {
+		final int numNodes = 7;
+		final long time = 1;
+		final TimeUnit timeUnit = TimeUnit.MINUTES;
+		final long rngSeed = System.currentTimeMillis();
+		// probability that a single node out of all correct nodes will crash-stop
+		final double crashProbabilityPerSecond = 0.1;
+
+		final List<ECKeyPair> allNodes = createNodes(numNodes, "");
+		final Set<ECPublicKey> faultyNodesPubs = new HashSet<>();
+		final BFTTestNetwork bftNetwork = new BFTTestNetwork(allNodes);
+
+		// correct nodes should all get the same commits in the same order
+		//CHECKSTYLE:OFF
+		Observable<Object> correctCommitCheck = Observable.zip(
+				allNodes.stream()
+					.map(node -> bftNetwork.getVertexStore(node).lastCommittedVertex()
+						.map(vertex -> Pair.of(node, vertex)))
+					.collect(Collectors.toList()),
+				Arrays::stream)
+			.map(nodesAndVertices -> nodesAndVertices
+				// the following cast is fine, checkstyle, see above
+				.map(nodeAndVertex -> (Pair<ECPublicKey, Vertex>) nodeAndVertex)
+				.filter(nodeAndVertex -> !faultyNodesPubs.contains(nodeAndVertex.getFirst()))
+				.map(Pair::getSecond)
+				.distinct()
+				.collect(Collectors.toList()))
+			.doOnNext(committedVertices -> assertThat(committedVertices).hasSize(1))
+			.map(vertices -> vertices.get(0))
+			.map(o -> o);
+		//CHECKSTYLE:ON
+
```

Comment:
Would like a check to make sure that bft is always making progress here

Reply:
Good idea